sunpy/map/mapbase.py

```diff
@@ -3096,6 +3111,39 @@ def reproject_to(self, target_wcs, *, algorithm='interpolation', return_footprin
             raise ValueError(f"The specified algorithm must be one of: {list(functions.keys())}")
         func = functions[algorithm]
 
+        if auto_extent is not None:
```

Comment:
Could we move this to a helper function? I think that it could be useful for people to use with ndcube objects (i.e. use it to calculate adjustments to more complex WCSes).

and you know end up in astropy or reproject :stuck_out_tongue:

Reply:
Moved out to be `sunpy.util.extent_in_other_wcs()`

Reply:
This would definitely be useful in reproject!

Reply:
LETS GO UPSTREAM